### PR TITLE
Configure ClassStructure sniff to sort members by function instead of type

### PR DESCRIPTION
**Pull request by baci266**

### CODE REVIEW

**Review by fialamarek — APPROVED** [review]

File: src/Cdn77/ruleset.xml [hunk 1]
```diff
@@ -46,19 +46,19 @@
...
-                protected static abstract methods,
-                protected static methods,
-                private static methods,
                 constructor,
                 static constructors,
                 destructor,
                 magic methods,
+                public static abstract methods,
                 public abstract methods,
+                public static methods,
```
Comment (p4veI) [inline src/Cdn77/ruleset.xml:55]: prohodil bych 54 a 55

File: src/Cdn77/ruleset.xml [hunk 2]
```diff
@@ -46,19 +46,19 @@
...
                 public methods,
+                protected static abstract methods,
                 protected abstract methods,
+                protected static methods,
```
Comment (p4veI) [inline src/Cdn77/ruleset.xml:59]: prohodil bych 58 a 59

**Review by p4veI — CHANGES_REQUESTED** [review]

File: src/Cdn77/ruleset.xml (hunk 1 above)
Comment (simPod) [reply src/Cdn77/ruleset.xml:55]: why?

Comment (simPod) [reply src/Cdn77/ruleset.xml:55]: let's keep contract first

**Review by simPod — APPROVED** [review]

File: src/Cdn77/ruleset.xml (hunk 1 above)
Comment (p4veI) [reply src/Cdn77/ruleset.xml:55]: it's just like my opinion but for me this order seems logical.. keeping static together and non-static together. I know we have tools for this, but I think this order is quite straight forward, when "trying to figure out" where to place the method.

public/protected/private static abstract
public/protected/private static

public/protected/private abstract
public/protected/private

but I can live with both versions for sure.

Comment (simPod) [reply src/Cdn77/ruleset.xml:55]: So we keep contract (abstract) first, right?

Comment (p4veI) [reply src/Cdn77/ruleset.xml:55]: yeah, but grouped as they were until now basically.

**Review by p4veI — APPROVED** [review]

**Pull request merged**